stonesoup/initiator/simple.py

```diff
@@ -49,20 +59,87 @@ def initiate(self, detections, timestamp, **kwargs):
             A list of new tracks with an initial :class:`~.GaussianState`
         """
 
-        updater = ExtendedKalmanUpdater(self.measurement_model)
-
         tracks = set()
         for detection in detections:
-            measurement_prediction = updater.predict_measurement(
+            measurement_prediction = self.updater.predict_measurement(
                 self.prior_state, detection.measurement_model)
-            track_state = updater.update(SingleHypothesis(
+            track_state = self.updater.update(SingleHypothesis(
                 self.prior_state, detection, measurement_prediction))
             track = Track([track_state])
             tracks.add(track)
 
         return tracks
 
 
+class SinglePointMeasurementInitiator(SinglePointInitiator):
+    """SinglePointMeasurementInitiator class
+
+    This uses an :class:`~.Updater` to carry out an update using
+    provided :attr:`prior_state` for each unassociated detection, using the
+    measurements state vector in state space to replace the prior state vector.
+    """
+    skip_non_reversible: bool = Property(default=False)
+
+    def initiate(self, detections, timestamp, **kwargs):
+        """Initiates tracks given unassociated measurements
+
+        Parameters
+        ----------
+        detections : set of :class:`~.Detection`
+            A list of unassociated detections
+        timestamp: datetime.datetime
+            Current timestamp
+
+        Returns
+        -------
+        : set of :class:`~.Track`
+            A list of new tracks with an initial :class:`~.GaussianState`
+        """
+
+        tracks = set()
+        for detection in detections:
+            if detection.measurement_model is not None:
+                measurement_model = detection.measurement_model
+            else:
+                if self.measurement_model is None:
+                    raise ValueError("No measurement model specified")
+                else:
+                    measurement_model = self.measurement_model
+
+            if isinstance(measurement_model, LinearModel):
+                model_matrix = measurement_model.matrix()
+                inv_model_matrix = np.linalg.pinv(model_matrix)
+                state_vector = inv_model_matrix @ detection.state_vector
+            else:
+                if isinstance(measurement_model, ReversibleModel):
+                    try:
+                        state_vector = measurement_model.inverse_function(detection)
+                    except NotImplementedError:
+                        if not self.skip_non_reversible:
+                            raise
+                        else:
+                            continue
+                elif self.skip_non_reversible:
+                    continue
+                else:
+                    raise Exception("Invalid measurement model used.\
+                                    Must be instance of linear or reversible.")
```

Comment:
in this current implementation, a user wouldn't be able to make use of a reversible measurement model they would pass through to the initiator if the detections had a non-reversible measurement model attached to them, e.g. `CartesianToElevationBearingRange `and `CartesianToElevationBearingRangeRate`. I suppose in that instance a mapping would also be needed to access the correct components in the detection for the reversible measurement model to map back to state space, so things will probably get overly complicated and messy for a "simple" initiator. Happy with the current implementation.